src/LogCalls.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\PsrLogTestDoubles;
+
+/**
+ * Immutable and ordered collection of LogCall objects
+ *
+ * @since 2.0
+ *
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class LogCalls implements \IteratorAggregate {
+
+	private $calls;
+
+	/**
+	 * @param LogCall[] $calls
+	 */
+	public function __construct( LogCall... $calls ) {
+		$this->calls = $calls;
+	}
+
+	public function getIterator() {
+		return new \ArrayIterator( $this->calls );
+	}
+
+	/**
+	 * @return string[]
+	 */
+	public function getMessages(): array {
+		return array_map(
+			function( LogCall $logCall ) {
+				return $logCall->getMessage();
+			},
+			$this->calls
+		);
+	}
+
+	/**
+	 * @return LogCall|null
+	 */
+	public function getFirstCall() {
+		return empty( $this->calls ) ? null : $this->calls[0];
```

Comment:
Is this array guaranteed to be numerically indexed, starting with 0? I assume the `LogCall... $calls` syntax does guarantee this, but I'm not familiar enough with it.